packages/adapters/ethers/src/client.ts

```diff
@@ -696,4 +703,11 @@ export class EthersAdapter extends AdapterBlueprint {
       params: [params]
     })
   }
+
+  private getPreferredAccountType() {
+    return (
+      AccountController.state.preferredAccountType ||
+      OptionsController.state.defaultAccountTypes.eip155
+    )
+  }
```

Comment:
I think we should move this function to adapter blueprint and eacy adapter should get the preferred account type like `ChainController.getAccountProp(this.namespace).preferredAccountType` this will be more generic approach

```suggestion
  private getPreferredAccountType() {
    return (
      ChainController.getAccountProp(this.namespace).preferredAccountType ||
     OptionsController.state.defaultAccountTypes[this.namespace]
    )
  }
```

Reply:
Even more we should save the latest account type to the local storage for each namespace and get the value from there - I've did this in one of my local branches, you can skip and I could implement after we merge this